lang/scala/scala.py

```diff
@@ -0,0 +1,284 @@
+        actions.skip()
+
+    def code_block():
+        actions.insert('{}')
+        actions.key('left enter')
+
+    def code_operator_lambda():
+        actions.auto_insert(" => ")
+
+    def code_operator_subscript():
+        actions.insert("()")
+        actions.key("left")
+
+    def code_operator_assignment():
+        actions.auto_insert(" = ")
+
+    def code_operator_subtraction():
+        actions.insert(" - ")
+
+    def code_operator_subtraction_assignment():
+        actions.auto_insert(" -= ")
+
+    def code_operator_addition():
+        actions.auto_insert(" + ")
+
+    def code_operator_addition_assignment():
+        actions.auto_insert(" += ")
+
+    def code_operator_multiplication():
+        actions.auto_insert(" * ")
+
+    def code_operator_multiplication_assignment():
+        actions.auto_insert(" *= ")
+
+    def code_operator_exponent():
+        actions.auto_insert(" ^ ")
+
+    def code_operator_division():
+        actions.auto_insert(" / ")
+
+    def code_operator_division_assignment():
+        actions.auto_insert(" /= ")
+
+    def code_operator_modulo():
+        actions.auto_insert(" % ")
+
+    def code_operator_modulo_assignment():
+        actions.auto_insert(" %= ")
+
+    def code_operator_equal():
+        actions.auto_insert(" == ")
+
+    def code_operator_not_equal():
+        actions.auto_insert(" != ")
+
+    def code_operator_greater_than():
+        actions.auto_insert(" > ")
+
+    def code_operator_greater_than_or_equal_to():
+        actions.auto_insert(" >= ")
+
+    def code_operator_less_than():
+        actions.auto_insert(" < ")
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(" <= ")
+
+    def code_operator_and():
+        actions.auto_insert(" && ")
+
+    def code_operator_or():
+        actions.auto_insert(" || ")
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(" & ")
+
+    def code_operator_bitwise_and_assignment():
+        actions.auto_insert(' &= ')
+
+    def code_operator_increment():
+        actions.auto_insert('++')
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(" | ")
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(" ^ ")
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(" << ")
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(" <<= ")
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(" >> ")
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(" >>= ")
+
+    def code_self():
+        actions.auto_insert("this")
+
+    def code_insert_null():
+        actions.auto_insert("null")
+
+    def code_insert_is_null():
+        actions.auto_insert(" == null")
+
+    def code_insert_is_not_null():
+        actions.auto_insert(" != null")
+
+    def code_state_if():
+        actions.insert("if () ")
+        actions.key("left")
```

Comment:
Should probably use `edit.left()` here and below instead, in case the editor doesn't use arrow keys to move.